gbpservice/nfp/configurator/lib/schema_validator.py

```diff
@@ -0,0 +1,77 @@
+from oslo_log import log
+import gbpservice.nfp.configurator.lib.schema as schema
+
+LOG = log.getLogger(__name__)
+
+
+class SchemaValidator():
+
+    def decode(self, request_data):
+        try:
+            if not self.validate_schema(request_data, schema.request_data):
+                return False
+
+            if ('service_type' in request_data['info'] and
+                    'version' in request_data['info']):
+                service_type = request_data['info']['service_type']
+            elif ('service_type' not in request_data['info'] and
+                    'version' in request_data['info']):
+                service_type = 'generic'
+            elif not self.validate_schema(request_data['info'],
+                                          schema.request_data_info):
+                return False
+
+            for config in request_data['config']:
+                if not self.validate_schema(config,
+                                            schema.request_data_config):
+                    return False
+
+                resource_type = config['resource']
+                resource = config['kwargs']
```

Comment:
For generic config, the data comes inside kwargs. Are these parameters inside kwargs validated. Here we are getting kwargs from config dict. Inside this, there will be one more kwargs.